src/dftbp/timedep/linrespcommon.F90

```diff
@@ -411,469 +406,464 @@ subroutine getSqrOcc(occNr, win, nmatup, nmat, getIA, tSpin, n_ij)
+    !> End index of current rank in global RPA vectors
+    integer, intent(in) :: fGlobal
 
     !> Environment settings
     type(TEnvironment), intent(inout) :: env
 
-    !> Dense matrix descriptor
-    type(TDenseDescr), intent(in) :: denseDesc 
-
-    !> Overlap times eigenvector (nOrb, nOrb)
-    real(dp), intent(in) :: ovrXev(:,:,:)
-
-    !> Eigenvectors (nOrb, nOrb)
-    real(dp), intent(in) :: grndEigVecs(:,:,:)
+    !> Data type for atomic orbital information
+    type(TOrbitals), intent(in) :: orb
+     
+    !> Data structure for linear response
+    type(TLinResp), intent(in) :: lr
 
-    !> Occupation numbers
-    real(dp), intent(in) :: occNr(:,:)
+    !> Run time parameters of the Casida routine
+    type(TCasidaParameter), intent(in) :: rpa
 
-    !> Square root of occupation difference between vir and occ states
-    real(dp), intent(in) :: sqrOccIA(:)
+    !> machinery for transition charges between single particle levels
+    type(TTransCharges), intent(in) :: transChrg
+    
+    !> symmetry flag (singlet or triplet)
+    character, intent(in) :: sym
 
-    !> DFTB gamma matrix (nAtm, nAtom)
-    real(dp), intent(in) :: gamma(:,:)
+    !> Dense matrix descriptor
+    type(TDenseDescr), intent(in) :: denseDesc
 
-    !> Chemical species of the atoms
+    !> chemical species of the atoms
     integer, intent(in) :: species0(:)
 
-    !> Ground state spin constants for each species
-    real(dp), intent(in) :: spinW(:)
+    !> overlap times eigenvector. (nOrb, nOrb) [distributed]
+    real(dp), intent(in) :: ovrXev(:,:,:)
+    
+    !> eigenvectors (nOrb, nOrb) [distributed]
+    real(dp), intent(in) :: grndEigVecs(:,:,:)
 
-    !> Onsite matrix elements for shells (elements between s orbitals on the same shell are ignored)
-    real(dp), intent(in), allocatable :: onsMEs(:,:,:,:)
+    !> DFTB gamma matrix (nAtm, nAtom)
+    real(dp), intent(in) :: frGamma(:,:)
 
-    !> Data type for atomic orbital information
-    type(TOrbitals), intent(in) :: orb
+    !> long-range Gamma if in use
+    real(dp), allocatable, optional, intent(in) :: lrGamma(:,:)
 
-    !> Whether (A+B)v or (Omega)v should be returned
+    !> whether (A+B)v or (Omega)v should be returned
     logical, intent(in) :: tAplusB
 
-    !> Machinery for transition charges between single particle levels
-    type(TTransCharges), intent(in) :: transChrg
-
-    !> Vector to multiply with size(nmat)
+    !> vector to multiply with size(nMat)
     real(dp), intent(in) :: vin(:)
 
-    !> Vector containing the result on exit size(nmat)
+    !> vector containing the result on exit size(nMat)
     real(dp), intent(out) :: vout(:)
 
-    !> Is calculation range-separated?
-    logical, intent(in) :: tHybridXc
-
-    !> Long-range Gamma if in use
-    real(dp), allocatable, optional, intent(in) :: lrGamma(:,:)
-
-    integer :: nmat, natom, norb, ia, nxvv_a, nSpin
+    integer :: nMat, nOrb, ia, nxvv_a
     integer :: ii, aa, ss, jj, bb, ias, jbs, abs, ijs, jas, ibs
```

Comment:
`abs` is a somehwat unlucky name as it shadows the intrinsic `abs()` function... But I can live with it, if it is the natural logical name at this point.